### PR TITLE
Documented information about sleep and wake namespace

### PR DESCRIPTION
**Pull request by NitishKumar06**
Fixes: https://github.com/okteto/docs/issues/364

### CODE REVIEW

**Review by pchico83 — COMMENTED** [review]
@NitishKumar06 this is great! could you also update the subcommands available for `okteto namespace` here?
https://www.okteto.com/docs/reference/cli/#namespace

**Comment by NitishKumar06** [comment]
Hi @pchico83, the changes have been pushed.

**Review by NitishKumar06 — COMMENTED** [review]
A minor point has been mentioned.

**Comment by NitishKumar06** [comment]
This PR is ready to be merged now!

File: src/content/reference/cli.mdx
Comment (pchico83) [reply src/content/reference/cli.mdx:443]: Could we make it optional? that way, `okteto namespace sleep` sleeps the current namespace in my okteto context

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:451]: does the cli wai for the sleep operation to finish?

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:464]: does the cli wai for the wake operation to finish?

File: src/content/reference/cli.mdx
Comment (NitishKumar06) [reply src/content/reference/cli.mdx:451]: It waits for 1-2 sec.

File: src/content/reference/cli.mdx
Comment (NitishKumar06) [reply src/content/reference/cli.mdx:464]: It waits for 1-2 sec.

**Comment by NitishKumar06** [comment]
>Could we make it optional? that way, okteto namespace sleep sleeps the current namespace in my okteto context

Did you mean to exclude mentioning about okteto sleep? If No, we can add more information about okteto sleep mentioning something like "it sleeps the active namespace...".

**Comment by pchico83** [comment]
> > Could we make it optional? that way, okteto namespace sleep sleeps the current namespace in my okteto context
> 
> Did you mean to exclude mentioning about okteto sleep? If No, we can add more information about okteto sleep mentioning something like "it sleeps the active namespace...".

@NitishKumar06 I mean that `okteto namespace sleep` and `okteto namespace wake` should work without specifying the namespace name. In that case, we should sleep|wake the current namespace. We can get the current namespace from the okteto context with `okteto.Context().Namespace`

**Comment by NitishKumar06** [comment]
> @NitishKumar06 I mean that `okteto namespace sleep` and `okteto namespace wake` should work without specifying the namespace name. In that case, we should sleep|wake the current namespace. We can get the current namespace from the okteto context with `okteto.Context().Namespace`

That's correct. However, we've configured `sleep` and `wake` in such a way that both of these commands requires at least 1 argument to successfully execute. I think we'll need to raise a new PR at `okteto/okteto` for this addition as well. WDYT?

**Comment by pchico83** [comment]
> > @NitishKumar06 I mean that `okteto namespace sleep` and `okteto namespace wake` should work without specifying the namespace name. In that case, we should sleep|wake the current namespace. We can get the current namespace from the okteto context with `okteto.Context().Namespace`
> 
> That's correct. However, we've configured `sleep` and `wake` in such a way that both of these commands requires at least 1 argument to successfully execute. I think we'll need to raise a new PR at `okteto/okteto` for this addition as well. WDYT?

@NitishKumar06 yes, I would add that functionality to okteto/okteto before documenting it. It's how most people will use that command, and I think it's better to educate users that way from the beginning.

**Comment by NitishKumar06** [comment]
Hi @pchico83, I've added this functionality via [this PR](https://github.com/okteto/okteto/pull/3574). We're good to go with documenting it now.

File: src/content/reference/cli.mdx
Comment (NitishKumar06) [reply src/content/reference/cli.mdx:451]: Do we need to make any changes?

**Comment by NitishKumar06** [comment]
After the successful merge of [PR](https://github.com/okteto/okteto/pull/3574), are we ready to proceed with next round of review?

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:446]:
```suggestion
$ okteto namespace sleep [name]
```

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:443]:
```suggestion
Sleeps an okteto namespace. By default, it sleeps the current okteto namespace. 
```

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:454]:
```suggestion
If you'd like to sleep a namespace other than the current one, you can provide that namespace name as an argument to the command.
```

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:459]:
```suggestion
Wakes an okteto namespace. By default, it wakes the current okteto namespace.
```

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:462]:
```suggestion
$ okteto namespace wake [name]
```

File: src/content/reference/cli.mdx
Comment (pchico83) [inline src/content/reference/cli.mdx:470]:
```suggestion
If you'd like to wake a namespace other than the current one, you can provide that namespace name as an argument to the command.
```

**Comment by pchico83** [comment]
@rberrelleza @omnomagonz could you review this PR?

File: src/content/cloud/namespaces.mdx
Comment (omnomagonz) [inline src/content/cloud/namespaces.mdx:36]:
```suggestion
One of the best ways to reduce cost while working is to put your namespace in the `sleep` state. Sleeping a namespace scales your development environment down to an idle state to minimize your costs to the absolute minimum, savings dozens of hours per week in cloud costs.

You can specify a global, default threshold for automatically sleeping a namespace in the [garbage collection](/self-hosted/administration/configuration/#gc) configuration.
```

@pchico83 I assume this wording is appropriate since the costs won't be "zero". Or would they? I'm unclear on the specifics of whether it's something on our end that we wake the environment when needed and/or if the environment leaves some bare minimum processes in place for the wake operation.

In either case: @NitishKumar06 this is a non-blocking proposed change 😄 I'm proposing this language (and proposing we remove the text below) since docs are typically direct and objective and this copy comes across as more appropriate for a marketing page to promote the value of this feature (sleep/wake)

File: src/content/cloud/namespaces.mdx
Comment (omnomagonz) [inline src/content/cloud/namespaces.mdx:37]:
```suggestion
```

File: src/content/cloud/namespaces.mdx
Comment (omnomagonz) [inline src/content/cloud/namespaces.mdx:38]:
```suggestion
```

File: src/content/cloud/namespaces.mdx
Comment (omnomagonz) [inline src/content/cloud/namespaces.mdx:46]:
```suggestion
In order to wake the namespace, you can click on Wake dialog.

:::tip

You can specify a global, default threshold for automatically sleeping a namespace in the [garbage collection](/self-hosted/administration/configuration/#gc) configuration.

:::

```

File: src/content/cloud/namespaces.mdx
Comment (pchico83) [reply src/content/cloud/namespaces.mdx:36]: I agree. I would reduce the scope of the PR to document the CLI commands

Comment (NitishKumar06) [reply src/content/cloud/namespaces.mdx]: Thanks! I've removed the text.

File: src/content/reference/cli.mdx
Comment (pchico83) [reply src/content/reference/cli.mdx:450]: No, not here

File: src/content/cloud/namespaces.mdx
Comment (omnomagonz) [reply src/content/cloud/namespaces.mdx:42]: @NitishKumar06 Please 😄

If you commit the suggestion I think we're set on this one 👍

Comment (NitishKumar06) [reply src/content/cloud/namespaces.mdx]: completely missed this one! Thanks for reminding me - Changes are applied!

File: src/content/cloud/namespaces.mdx
Comment (pchico83) [inline src/content/cloud/namespaces.mdx:36]:
```suggestion
You can sleep your namespace from the command line with `okteto namespace sleep [name]`.
```
That means it's optional

File: src/content/cloud/namespaces.mdx
Comment (pchico83) [inline src/content/cloud/namespaces.mdx:38]:
```suggestion
When you want to wake your namespace, you can use the command `okteto namespace wake [name]`.
```
That means it's optinal

**Review by pchico83 — APPROVED** [review]

File: src/content/cloud/namespaces.mdx
Comment (NitishKumar06) [reply src/content/cloud/namespaces.mdx]: Done!

**Comment by pchico83** [comment]
Thanks @NitishKumar06 for taking care of documenting the feature!

**Comment by NitishKumar06** [comment]
> Thanks @NitishKumar06 for taking care of documenting the feature!

My pleasure! I'd like to inform the community about this update. How about sending a message on slack mentioning about the docs we have updated? Or shall we wait if needed?

**Comment by pchico83** [comment]
@NitishKumar06 I would wait until the next official CLI release. I think that's the one including the latest fixes.
After merging the docs, they don't go to the main page until we do the next release.
The go to "latest":
https://www.okteto.com/docs/1.8/reference/cli

**Comment by NitishKumar06** [comment]
> @NitishKumar06 I would wait until the next official CLI release. I think that's the one including the latest fixes. After merging the docs, they don't go to the main page until we do the next release. They go to "latest": https://www.okteto.com/docs/1.8/reference/cli

I see. Do we have dates we're targeting for the next okteto CLI release?

**Comment by pchico83** [comment]
> > @NitishKumar06 I would wait until the next official CLI release. I think that's the one including the latest fixes. After merging the docs, they don't go to the main page until we do the next release. They go to "latest": https://www.okteto.com/docs/1.8/reference/cli
> 
> I see. Do we have any dates we're targeting for the next okteto CLI release?

Around 2 weeks from now

**Comment by NitishKumar06** [comment]
> Around 2 weeks from now

Awesome! Looking forward to contributing more!

**Pull request merged**
